src/node/blockstorage.cpp

```diff
@@ -1067,6 +1067,11 @@ bool BlockManager::ReadBlockFromDisk(CBlock& block, const CBlockIndex& index) co
 bool BlockManager::ReadRawBlockFromDisk(std::vector<uint8_t>& block, const FlatFilePos& pos) const
 {
     FlatFilePos hpos = pos;
+    // If nPos is less than 8 we don't have the block data and can't seek back to get meta header
+    // Return early to prevent undefined behavior of unsigned int underflow
+    if (hpos.nPos < 8) {
+        return error("%s: OpenBlockFile failed for %s", __func__, pos.ToString());
+    }
```

Comment:
This check seems like the right thing to do, but this could be handled by the caller, by checking `if (!(pindex->nStatus & BLOCK_HAVE_DATA))`, like done in `net_processing.cpp:2292`. Maybe still add this in a separate commit to clearly explain what is going and why being defensive here is an improvement?

Reply:
Done, added as a separate commit at the start of this PR.
Yes, it could be handled by the caller, but it's really handled by `OpenBlockFile` which checks for a null `pos`. It's just that before that check it blindly decrements the nPos by 8. Now we check that and return the same error that would be returned by `OpenBlockFile` if `pos` is null.